arch/xtensa/core/xtensa_mmu.c

```diff
@@ -485,6 +486,29 @@ void z_xtensa_mmu_smp_init(void)
 	xtensa_mmu_init(false);
 }
 
+#ifdef CONFIG_ARCH_HAS_RESERVED_PAGE_FRAMES
+/* Zephyr's linker scripts for Xtensa usually puts
+ * something before z_mapped_start (aka .text),
+ * i.e. vecbase, so that we need to reserve those
+ * space or else k_mem_map() would be mapping those,
+ * resulting in faults.
+ */
```

Comment:
This seems like the wrong spot for this.  "Area between CONFIG_SRAM_BASE_ADDRESS and z_mapped_start is reserved" is a fine rule, but if it's a rule it's a rule at the SOC layer somewhere and not the arch, at least as we're doing linking currently.  Maybe delegate to a z_soc_reserve_pages() or something and put this code there, where you can be more assertive than "linker scripts usually put" (e.g. "On TGL/ACE the vectors and iram section live below SRAM, reserve them", along with your own linker-emitted addresses on exactly the boundary).

Note that the mtk_adsp setup (which to be fair doesn't have an MMU) links most of its stuff to a DRAM region and not above SRAM_BASE_ADDRESS, puts vecbase at an arbitrary address (or will in the next version).  There's likely to be a lot of variation.